Use entry_points instead of scripts for cross-platform compatibility

akx opened the pull request:
This makes `tx` work right out of the box when you `pip install transifex-client` on Windows.

akx commented:
Looks like CircleCI builds are just simply broken, timing out asking for a password?

> Please enter your transifex username:
> command tox -v --recreate took more than 10 minutes since last output

akx commented:
This is something of a duplicate of #69, it seems.

diegobz commented:
@akx It looks like the Circleci issue is related to your PR. Just rebuild master and it passed fine. https://circleci.com/gh/transifex/transifex-client/78

akx commented:
@diegobz: The same issue happens with my other PR, #104...

akx commented:
@diegobz: I believe it's a problem with the "Exporting env vars from project settings" step not happening for builds that have code not owned by a project maintainer.

> Exporting env vars from project settings...
> > Exporting TRANSIFEX_PASSWORD
> > Exporting TRANSIFEX_USER

diegobz commented:
@akx alright, I will try to check it out. In the meanwhile, could you make sure to update the CI tests for windows to also take into account the `pip install` scenario?

akx commented:
@diegobz: Sure, I can try.

Also -- the env var thing is a good thing from a security viewpoint ([Travis-CI's documentation explains the same concept](https://docs.travis-ci.com/user/pull-requests#Security-Restrictions-when-testing-Pull-Requests)) but problematic for this stuff. (The suggested workaround is to skip tests that require secrets in PR builds.)

However, I'm now sort of concerned about the fact that AppVeyor builds the same code correctly, giving me reason to believe that it injected those envvars correctly, thus allowing anyone who can manage to trick it to print environment variables to the log to expose these secrets... :speak_no_evil:

akx commented:
@diegobz: The Windows build now fails because a non-PyInstallered tx script requires the SSL fix from #104 (the path to the bundled certificate is wrong here), but at least it demonstrably attempts to run both scripts... :)

akx commented:
@diegobz: Done. I also added a check against missing credentials so PR builds should now succeed (though with less test coverage).

diegobz commented on contrib/tx_commands.sh:
```diff
@@ -1,3 +1,8 @@
+if [ -z "$TRANSIFEX_USER" ]; then
```
Is there a way to identify a build from PRs instead? I'm afraid checking for `$TRANSIFEX_USER` may hide an misconfiguration in case we forget to set it up in CircleCI, or any other CI tool we may wan to use in the feature.

diegobz commented:
@akx I'm thinking `txclib/cmdline.py` should probably be called `tx.py`. It seems more intuitive as the CLI is called `tx`. Testing it directly from the repo would be just a `tx.py --help`, for example. However I'm not sure whether there is any standard about it that we should be following.

akx commented:
@diegobz:

* Switched to CircleCI's `CI_PULL_REQUEST` envvar for the check
* Added a `__main__.py`, so testing the client directly from the repo is now `python -m txclib ...`. I felt calling it `tx.py` was a little weird, but if you disagree, I can just change it :)

akx commented:
Okay, that's CircleCI finally tamed... Didn't realize Tox doesn't pass all envvars through by default :)

akx commented:
Ping @diegobz, could this be merged?

diegobz commented:
@akx I will bump this in our TODO. Sorry for the late reply.

kouk commented:
@akx @diegobz I'm fine with this, except for one modification I will push after merging. See https://github.com/transifex/transifex-client/commit/40aa093d78bf8947e89032c43bca0ab122be5331
